### PR TITLE
CCM-7651: CIS2 backend

### PR DESCRIPTION
**Pull request by mark-r-bjss**
## Type of changes

- [ ] Refactoring (non-breaking change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [ ] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [ ] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

**Review by bhansell1 — APPROVED** [review]

File: infrastructure/terraform/components/app/cognito_user_pool_domain_custom.tf [hunk 1]
```diff
@@ -1,6 +1,8 @@
 resource "aws_cognito_user_pool_domain" "custom" {
+  count = local.use_custom_cognito_domain ? 1 : 0
+
   user_pool_id    = aws_cognito_user_pool.main.id
-  certificate_arn = aws_acm_certificate.cognito.arn
+  certificate_arn = aws_acm_certificate.cognito[0].arn
```
Comment (aidenvaines-bjss) [inline infrastructure/terraform/components/app/cognito_user_pool_domain_custom.tf:5]: making this conditional is a good shout

File: infrastructure/terraform/components/app/module_amplify_branch.tf [hunk 2]
```diff
@@ -1,14 +1,14 @@
 module "amplify_branch" {
   source            = "git::https://github.com/NHSDigital/nhs-notify-shared-modules.git//infrastructure/modules/amp_branch?ref=v1.0.0"
-  name              = "main"
+  name              = var.branch_name
```
Comment (aidenvaines-bjss) [inline infrastructure/terraform/components/app/module_amplify_branch.tf:3]: Is production branch not always main? why the change?

**Review by aidenvaines-bjss — APPROVED** [review]

File: infrastructure/terraform/components/app/module_amplify_branch.tf (hunk 2 above)
Comment (mark-r-bjss) [reply infrastructure/terraform/components/app/module_amplify_branch.tf:3]: This allows you to override the production branch for occasions where you want to deploy your custom branch rather than main which is likely when testing a whole new dev environment.

**Review by aidenvaines-bjss — APPROVED** [review]

**Review by bhansell1 — APPROVED** [review]

**Review by gparry333 — APPROVED** [review]

**Review by gparry333 — APPROVED** [review]
